lib/perl/Genome/ModelDeprecated.pm

```diff
@@ -631,38 +631,53 @@ sub _resolve_type_name_for_class {
 
 sub compatible_instrument_data {
     my $self = shift;
-    my %params;
 
-    if (my @samples = $self->get_all_possible_samples)  {
-        my @sample_ids = map($_->id, @samples);
-        %params = (
-                   sample_id => \@sample_ids,
-               );
+    if($self->analysis_project) {
+        my $config_item = $self->analysis_project_bridge->config_profile_item;
+        unless($config_item) {
+            die $self->error_message('Configuration missing from analysis project association.  Cannot determine compatible instrument data for model %s and analysis project %s.', $self->__display_name__, $self->analysis_project->__display_name__);
+        }
+
+        my $rmm = Genome::Config::Translator->get_rule_model_map_from_config($config_item);
+        my @project_instrument_data = $self->analysis_project->instrument_data;
```

Comment:
I don't think converting the rule into the BoolExpr wouldn't solve the trade-off problem you've brought up but it would obviously be more performant than doing the matching client side.

CQID could be using `AutoUnloadPool` around `_process_models` or something.  Do we not have memory problems with CQID now?  It looks like the old code would load everything as well?